### PR TITLE
Add and use defines for camera backends

### PR DESCRIPTION
**Pull request by peterbarker**
```
Board              AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                      *      *           *       *                 *      *      *
Hitec-Airspeed     *                                                                     
KakuteH7-bdshot               *      *           *       *                 *      *      *
MatekF405                     *      *           *       *                 *      *      *
Pixhawk1-1M                   8      *           -88     -88               -88    -88    -88
f103-QiotekPeriph  *                                                                     
f303-Universal     0                                                                     
iomcu                                                          *                         
revo-mini                     *      *           *       *                 *      *      *
skyviper-v2450                                   -760                                    
```

### CODE REVIEW

**Comment by peterbarker** [comment]
Removed the Mount backend from minimized boards, thus the size savings there.

File: libraries/AP_Camera/AP_Camera.h [hunk 1]
```diff
@@ -47,11 +47,21 @@ class AP_Camera {
     // enums
     enum class CameraType {
         NONE = 0,           // None
+#if AP_CAMERA_SERVO_ENABLED
```
Comment (rmackay9) [inline libraries/AP_Camera/AP_Camera.h:50]: I don't think there's any reason to get rid of the enums..

File: libraries/AP_HAL_ChibiOS/hwdef/include/minimize_features.inc [hunk 2]
```diff
@@ -52,3 +52,6 @@ define OSD_PARAM_ENABLED 0
 
 # no Winch if minimized:
 define AP_WINCH_ENABLED 0
+
+# prune out some odd camera backends:
+define AP_CAMERA_MOUNT_ENABLED 0
```
Comment (rmackay9) [inline libraries/AP_HAL_ChibiOS/hwdef/include/minimize_features.inc:57]: this is fine but we could also make the ap-camera-mount-enabled depend upon ap-camera && hal-mount-enabled.  Just an idea though.

File: libraries/AP_Camera/AP_Camera.cpp [hunk 3]
```diff
@@ -86,31 +86,40 @@ void AP_Camera::init()
         CameraType camera_type = get_type(instance);
 
         // check for servo camera
-        if (camera_type == CameraType::SERVO) {
+        if (false) {
```
Comment (rmackay9) [inline libraries/AP_Camera/AP_Camera.cpp:89]: Txs for this.  This part is a bit awkward, could we make it like AP_Mount perhaps where we simply don't create the backend?

I plan to eventually add a healthy() and a pre-arm check.

**Comment by rmackay9** [comment]
In general this looks really good to me, thanks very much!

File: libraries/AP_HAL_ChibiOS/hwdef/include/minimize_features.inc (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/include/minimize_features.inc:57]: I think we did - so yes, this bit is redundant *except* if you add mount back in then you still won't get this backend- which I think is fine?

File: libraries/AP_Camera/AP_Camera.cpp (hunk 3 above)
Comment (peterbarker) [reply libraries/AP_Camera/AP_Camera.cpp:89]: I have a cleanup for this code which moves it to a switch statement; I was trying to keep this patch simple.

That other branch also saves ~260 bytes, so expect to see it soon!

File: libraries/AP_Camera/AP_Camera.h (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Camera/AP_Camera.h:50]: I *really* like getting rid of the enums!  When combined with switch() statements it really helps ensure you don't have stray code floating around which should be being excluded but isn't.  I've based a few PRs on this fact in the past.

File: libraries/AP_Camera/AP_Camera.cpp (hunk 3 above)
Comment (peterbarker) [reply libraries/AP_Camera/AP_Camera.cpp:89]: > Txs for this. This part is a bit awkward, could we make it like AP_Mount perhaps where we simply don't create the backend?
> 
> I plan to eventually add a healthy() and a pre-arm check.

I've now created a PR for that further cleanup - https://github.com/ArduPilot/ardupilot/pull/23127

I remembered why I *really* didn't want to mix in general cleanups - it means this PR has a whole lot of "*"s in the "bytes changed" information block in the opening comment for this PR - meaning that those builds are unchanged.  That definitely isn't the case after doing more cleanups.

File: libraries/AP_HAL_ChibiOS/hwdef/include/minimize_features.inc (hunk 2 above)
Comment (rmackay9) [reply libraries/AP_HAL_ChibiOS/hwdef/include/minimize_features.inc:57]: yes, it's fine to have a mount without the AP_Camera library..

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**